new-dti-website/components/courses/Timeline.tsx

```diff
@@ -0,0 +1,222 @@
+import React, { useLayoutEffect, useRef, useState } from 'react';
+
+// * TYPES
+export type Event = {
+  title: string;
+  date: string;
+  time: string;
+};
+
+// * INTERFACE
+type TimelineProps = {
+  events: Event[];
+  currentDate: Date;
+};
+
+/**
+ * `Timeline` Component - Displays a chronological timeline of Trends events.
```

Comment:
I think as a team we haven't put a lot of emphasis on documenting the helpers or components. I think it's super helpful for the util functions. For components I think it's optional but I don't mind at all that we have them here!